lib/Parse/ParseExpr.cpp

```diff
@@ -2905,9 +2913,49 @@ ExprResult Parser::ParseBoundsExpression() {
     SkipUntil(tok::r_paren, StopAtSemi | StopBeforeMatch);
 
   PT.consumeClose();
+
+  if (StartsRelativeBoundsClause(Tok)) {
```

Comment:
The reason I put this at the end of ParseBoundsExpression() is that PT.consumeClose() at the previous line. The PT is affected all the Bounds Expression and I think if it moves in the parsing of RangeBoundsExpr, it might not be natural (something like ... bounds(a, a+1 rel_align(char)). and have to add PT.consumeClose() for every BoundsExpression. So I adds check the Ident of Tok is kind of relative alignment caluase in StartRelativeBoundsClause() to do the parsing of relativeboundsclause.